### ISSUE TITLE
How to configure URLs of articles with underscore

### ISSUE THREAD

**Issue by molotoff** [middleman/middleman-blog#376]
## Expected behaviour
Get the articles in build with initially names with underscore

## Actual behaviour
Now all urls are replacing with dashes

## Steps to reproduce the problem
1. Create the article with underscore: `name_of_article.html.markdown`
2. After the building we get this article with dashes `name-of-article.html`

- Ruby version: 2.3.3
- Middleman version: 4.2.1
- Middleman Blog version: 4.0.3
- OS version: MacOS 10.15.2

**Comment by tdreyno** [comment]
Are you able to test on the current stable version 4.3.5?

**Comment by tdreyno** [comment]
Found some code in the blog extension that might explain this: https://github.com/middleman/middleman-blog/blob/master/lib/middleman-blog/uri_templates.rb#L76

**Comment by molotoff** [comment]
Yeah, I saw those part of code. 
But is fork of repository is a single way to fix those problem?

**Comment by tdreyno** [comment]
No, if we find a fix we can update this repository. The biggest issue is that this "bug" has been around for years so changing it now might change other users URLs. So we'd have to fix this by adding a new option to opt in the new (aka fixed) behavior.

**Comment by markets** [comment]
ℹ️ This issue is stale because it has been open for more than 90 days with no activity. It will be closed in 30 days if no further activity occurs. Thank you for your contributions.

**Comment by markets** [comment]
@tdreyno Do you know how can I access the blog options in that file? I just made a quick look, but I can't find any way to access the blog instance here:

https://github.com/middleman/middleman-blog/blob/9bb7fc71179d2da4227aa4649e3ec6e429c880c2/lib/middleman-blog/uri_templates.rb#L80-L81

I'd like to introduce a new option to allow to opt in this behavior.

**Comment by tdreyno** [comment]
@markets looks like a static helper function. probably best to see if there's access to the blog instance/config from wherever its being called. Can add params/options to that helper to be more configurable.

**Comment by markets** [comment]
Yeah, that's what I thought too.. Ok! I'll take a look taking in mind this approach.
